src/runtime/virtcontainers/stratovirt.go

```diff
@@ -0,0 +1,1306 @@
+		return "", fmt.Errorf("StratoVirt path (%s) does not exist", path)
+	}
+	return path, nil
+}
+
+// Log StratoVirt errors and ensure the StratoVirt process is reaped after
+// termination
+func (s *stratovirt) logAndWait(stratovirtCmd *exec.Cmd, reader io.ReadCloser) {
+	s.state.pid = stratovirtCmd.Process.Pid
+	s.Logger().Infof("Start logging StratoVirt (Pid=%d)", s.state.pid)
+	scanner := bufio.NewScanner(reader)
+	infoRE := regexp.MustCompile("([^:]):INFO: ")
+	warnRE := regexp.MustCompile("([^:]):WARN: ")
+	for scanner.Scan() {
+		text := scanner.Text()
+		if infoRE.MatchString(text) {
+			text = infoRE.ReplaceAllString(text, "$1")
+			s.Logger().WithField("StratoVirt Pid", s.state.pid).Info(text)
+		} else if warnRE.MatchString(text) {
+			text = infoRE.ReplaceAllString(text, "$1")
+			s.Logger().WithField("StratoVirt Pid", s.state.pid).Warn(text)
+		} else {
+			s.Logger().WithField("StratoVirt Pid", s.state.pid).Error(text)
+		}
+	}
+	s.Logger().Infof("Stop logging StratoVirt (Pid=%d)", s.state.pid)
+	stratovirtCmd.Wait()
+}
+
+// waitVM will wait for the Sandbox's VM to be up and running.
+func (s *stratovirt) waitVM(ctx context.Context, timeout int) error {
+	span, _ := katatrace.Trace(ctx, s.Logger(), "waitVM", stratovirtTracingTags, map[string]string{"sandbox_id": s.id})
+	defer span.End()
+
+	if timeout < 0 {
+		return fmt.Errorf("Invalid timeout %ds", timeout)
+	}
+
+	cfg := govmmQemu.QMPConfig{Logger: newQMPLogger()}
+
+	var qmp *govmmQemu.QMP
+	var disconnectCh chan struct{}
+	var ver *govmmQemu.QMPVersion
+	var err error
+
+	// clear andy possible old state before trying to connect again.
+	s.qmpShutdown()
+	timeStart := time.Now()
+	for {
+		disconnectCh = make(chan struct{})
+		qmp, ver, err = govmmQemu.QMPStart(s.qmpMonitorCh.ctx, s.qmpMonitorCh.path, cfg, disconnectCh)
+		if err == nil {
+			break
+		}
+
+		if int(time.Since(timeStart).Seconds()) > timeout {
+			return fmt.Errorf("Failed to connect StratoVirt instance (timeout %ds): %v", timeout, err)
+		}
+
+		time.Sleep(time.Duration(50) * time.Millisecond)
+	}
+	s.qmpMonitorCh.qmp = qmp
+	s.qmpMonitorCh.disconn = disconnectCh
+	defer s.qmpShutdown()
+
+	s.Logger().WithFields(logrus.Fields{
+		"qmp-major-version": ver.Major,
+		"qmp-minor-version": ver.Minor,
+		"qmp-micro-version": ver.Micro,
+		"qmp-Capabilities":  strings.Join(ver.Capabilities, ","),
+	}).Infof("QMP details")
+
+	if err = s.qmpMonitorCh.qmp.ExecuteQMPCapabilities(s.qmpMonitorCh.ctx); err != nil {
+		s.Logger().WithError(err).Error(qmpCapErrMsg)
+		return err
+	}
+
+	return nil
+}
+
+func (s *stratovirt) createParams(params *[]string) {
+	*params = append(*params, "-name", s.svConfig.name)
+	*params = append(*params, "-uuid", s.svConfig.uuid)
+	*params = append(*params, "-smp", strconv.Itoa(int(s.svConfig.smp)))
+	*params = append(*params, "-m", strconv.Itoa(int(s.svConfig.memory)))
+	*params = append(*params, "-kernel", s.svConfig.kernelPath)
+	*params = append(*params, "-append", s.svConfig.params)
```

Comment:
Seeing as `s.svConfig.params` becomes the value of `-append`, it might be better to call it something more descriptive - `kernelParams` or `kernelAdditionalParams` perhaps?  Seeing the current field definition with a very generic name (`params`) and a very generic type (`string`) I had difficulties initially to figure out what the field is for.

Reply:
The `s.svConfig.params` naming is inappropriate indeed, `kernelAdditionalParams` is good. Thanks for your advices and I'll take note of it~

Reply:
Thank you @WenyuanLau , you're very kind!